src/components/ConversationSettings/DangerZone.vue

```diff
@@ -0,0 +1,134 @@
+  -
+  - @license GNU AGPL version 3 or any later version
+  -
+  - This program is free software: you can redistribute it and/or modify
+  - it under the terms of the GNU Affero General Public License as
+  - published by the Free Software Foundation, either version 3 of the
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+-->
+
+<template>
+	<div>
+		<div class="app-settings-section__hint">
+			{{ t('spreed', 'Be careful, these actions cannot be undone.') }}
+		</div>
+		<button v-if="canLeaveConversation"
+			@click.prevent.exact="leaveConversation">
+			{{ t('spreed', 'Leave conversation') }}
+		</button>
+		<button v-if="canDeleteConversation"
+			class="critical error"
+			@click.prevent.exact="deleteConversation">
+			{{ t('spreed', 'Delete conversation') }}
+		</button>
+	</div>
+</template>
+
+<script>
+import { removeCurrentUserFromConversation } from '../../services/participantsService'
+import { showError } from '@nextcloud/dialogs'
+import { deleteConversation } from '../../services/conversationsService'
+import { emit } from '@nextcloud/event-bus'
+
+export default {
+	name: 'DangerZone',
+
+	props: {
+		conversation: {
+			type: Object,
+			required: true,
+		},
+
+		canLeaveConversation: {
+			type: Boolean,
+			required: true,
+		},
+
+		canDeleteConversation: {
+			type: Boolean,
+			required: true,
+		},
+	},
+
+	computed: {
+		token() {
+			return this.conversation.token
+		},
+	},
+
+	methods: {
+
+		hideConversationSettings() {
+			emit('hide-conversation-settings')
+		},
+		/**
+		 * Deletes the current user from the conversation.
+		 */
+		async leaveConversation() {
+			try {
+				await removeCurrentUserFromConversation(this.token)
+				// If successful, deletes the conversation from the store
+				this.$store.dispatch('deleteConversation', this.conversation)
+				this.hideConversationSettings()
+			} catch (error) {
+				if (error.response && error.response.status === 400) {
+					showError(t('spreed', 'You need to promote a new moderator before you can leave the conversation.'))
+				} else {
+					console.error(`error while removing yourself from conversation ${error}`)
+				}
+			}
+		},
+
+		/**
+		 * Deletes the conversation.
+		 */
+		async deleteConversation() {
+			OC.dialogs.confirm(
+				t('spreed', 'Do you really want to delete "{displayName}"?', this.conversation),
+				t('spreed', 'Delete conversation'),
+				async function(decision) {
+					if (!decision) {
+						return
+					}
+
+					if (this.token === this.$store.getters.getToken()) {
+						this.$router.push('/apps/spreed')
```

Comment:
optional: recently I found out that we can also use route names instead of paths
here it would be:
```
this.$router.push({ name: 'root' })
```

Reply:
done!